app/utils/data_access.py

```diff
@@ -64,29 +65,33 @@ def register_user(email, password, validated=False, privileges=[]):
         email=email,
         uuid=uuid.uuid4().hex,
         password_hash=generate_password_hash(password),
+        username=username,
         validated=validated,
         privileges=privileges,
     )
     with DATABASE.atomic():
         user.save()
     return user
 
-def authenticate_user(email, password):
+def authenticate_user(name, password):
     """
-    Authenticates an email/password pair against the database. Returns the
-    user info for the authenticated user on success, or None on failure.
+    Authenticates an email or username and password against the database.
+    Returns the user info for the authenticated user on success, or None on
+    failure.
     """
+    if not name or not password:
+        return None
     try:
         with DATABASE.atomic():
-            user_info = User.get(User.email == email)
+            user_info = User.select().where((User.email == name) | (User.username == name)).get()
```

Comment:
Can we get a problem if one user's username is another user's email address?

Reply:
yes - that will be solved later when I apply validation and rules, which will include emails needing an '@', and usernames disallowing special characters though.